paddle/fluid/inference/tests/api/analyzer_int8_resnet50_tester.cc

```diff
@@ -0,0 +1,106 @@
+/* Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <fstream>
+#include <iostream>
+#include "paddle/fluid/inference/tests/api/tester_helper.h"
+
+namespace paddle {
+namespace inference {
+namespace analysis {
+
+template <typename T>
+constexpr paddle::PaddleDType GetPaddleDType();
+
+template <>
+constexpr paddle::PaddleDType GetPaddleDType<int64_t>() {
+  return paddle::PaddleDType::INT64;
+}
+
+template <>
+constexpr paddle::PaddleDType GetPaddleDType<float>() {
+  return paddle::PaddleDType::FLOAT32;
+}
+
+void SetConfig(AnalysisConfig *cfg) {
+  cfg->SetModel(FLAGS_infer_model);
+  cfg->SetProgFile("__model__");
+  cfg->DisableGpu();
+  cfg->SwitchIrOptim();
+  cfg->SwitchSpecifyInputNames();
+  cfg->SetCpuMathLibraryNumThreads(FLAGS_paddle_num_threads);
+}
+
+template <typename T>
+PaddleTensor LoadTensorFromStream(std::ifstream &file, std::vector<int> shape,
+                                  std::string name) {
+  PaddleTensor tensor;
+  tensor.name = name;
+  tensor.shape = std::move(shape);
+  tensor.dtype = GetPaddleDType<T>();
+  size_t numel = std::accumulate(begin(tensor.shape), end(tensor.shape), 1,
+                                 std::multiplies<T>());
+  tensor.data.Resize(numel * sizeof(T));
+
+  std::copy_n(std::istream_iterator<T>(file), numel,
+              static_cast<T *>(tensor.data.data()));
+
+  return tensor;
+}
+
+void SetInput(std::vector<std::vector<PaddleTensor>> *inputs) {
+  PADDLE_ENFORCE_EQ(FLAGS_test_all_data, 0, "Only have single batch of data.");
+  std::string line;
+  std::ifstream file(FLAGS_infer_data);
+
+  int batch_size = 100;
+  PaddleTensor data =
+      LoadTensorFromStream<float>(file, {batch_size, 3, 224, 224}, "data");
+
+  PaddleTensor labels =
+      LoadTensorFromStream<int64_t>(file, {batch_size, 1}, "labels");
+
+  std::vector<PaddleTensor> input_slots{{data, labels}};
+  inputs->emplace_back(input_slots);
+}
+
+TEST(Analyzer_int8_resnet50, quantization) {
```

Comment:
This `TEST(Analyzer_int8_resnet50, quantization)` is only for `CompareQuantizedNativeAndAnalysis`, how could we do profiler?

Reply:
I will add profiler.